fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/Error3.scala

```diff
@@ -27,6 +27,63 @@ trait Error3[F[-_, +_, +_]] extends ApplicativeError3[F] with Monad3[F] {
     tap(tapError[R, E, A, E1](r)(err), succ)
   }
 
+  /**
+    * Retries this effect until its error satisfies the specified predicate.
+    */
+  def retryUntil[R, E, A](r: F[R, E, A])(f: E => Boolean): F[R, E, A] =
+    retryUntilM(r)(e => pure(f(e)))
+
+  /**
+    * Retries this effect until its error satisfies the specified effectful predicate.
+    */
+  def retryUntilM[R, E, A](r: F[R, E, A])(f: E => F[R, Nothing, Boolean]): F[R, E, A] =
+    catchAll(r) {
+      e =>
+        flatMap(f(e)) {
+          b =>
+            if (b) {
+              fail(e)
+            } else {
+              retryUntilM(r)(f)
+            }
+        }
+    }
+
+  /**
+    * Retries this effect while its error satisfies the specified predicate.
+    */
+  def retryWhile[R, E, A](r: F[R, E, A])(f: E => Boolean): F[R, E, A] =
+    retryWhileM(r)(e => pure(f(e)))
+
+  /**
+    * Retries this effect while its error satisfies the specified effectful predicate.
+    */
+  def retryWhileM[R, E, A](r: F[R, E, A])(f: E => F[R, Nothing, Boolean]): F[R, E, A] =
+    retryUntilM(r)(e => map(f(e))(!_))
+
+  /**
+    * Extracts the optional value, or returns the given 'default'.
+    */
+  def someOrElse[R, E, A](r: F[R, E, Option[A]])(default: => A): F[R, E, A] =
+    map(r)(_.getOrElse(default))
+
+  /**
+    * Extracts the optional value, or executes the effect 'default'.
+    */
+  def someOrElseM[R, E, A](r: F[R, E, Option[A]])(default: F[R, E, A]): F[R, E, A] =
```

Comment:
Could you please rename this to `fromOptionF` to align with existing `fromOption` method? Also I think it's better to suspend the `default` argument to be consistent with `ApplicativeError3#orElse`